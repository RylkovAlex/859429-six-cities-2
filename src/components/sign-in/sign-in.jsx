import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

import Header from '../header/header.jsx';
import withInputsState, {validator} from '../hocs/withInputsState/withInputsState';
import {Operation} from '../../redux/actions/action-creator/action-creator.js';

const SignIn = ({setInput, inputValues, logIn}) => {
  return (
    <div className="page page--gray page--login">
      <Header/>
      <main className="page__main page__main--login">

        <div className="page__login-container container">
          <section className="login">
            <h1 className="login__title">Sign in</h1>
            <form className="login__form form">
              <div className="login__input-wrapper form__input-wrapper">
                <label className="visually-hidden">E-mail</label>
                <input
                  className="login__input form__input"
                  type="email"
                  name="email"
                  placeholder="Email"
                  required=""
                  onChange = {(evt) => setInput(`email`, evt.target.value)}
                />
              </div>
              <div className="login__input-wrapper form__input-wrapper">
                <label className="visually-hidden">Password</label>
                <input
                  className="login__input form__input"
                  type="password"
                  name="password"
                  placeholder="Password"
                  required=""
                  onChange = {(evt) => setInput(`password`, evt.target.value)}
                />
              </div>
              <button
                className="login__submit form__submit button"
                type="submit"
                disabled = {!(inputValues.password && inputValues.email)}
                onClick = {(evt) => {
                  evt.preventDefault();
                  logIn(inputValues);
                }}
              >
                Sign in
              </button>
            </form>
          </section>
          <section className="locations locations--login locations--current">
            <div className="locations__item">
              <a className="locations__item-link" href="#">
                <span>Amsterdam</span>
              </a>
            </div>

          </section>
        </div>
      </main>
    </div>
  );
};

SignIn.propTypes = {
  setInput: PropTypes.func.isRequired,
  inputValues: PropTypes.object.isRequired,
  logIn: PropTypes.func.isRequired,
};

const mapDispatchToProps = (dispatch) => ({
  logIn: (authData) => dispatch(Operation.fetchAuthData(authData)),
});

export default connect(null, mapDispatchToProps)(
    withInputsState(
        SignIn,
        [`email`, `password`],
        {
          'email': [validator.EMAIL],
          'password': [validator.REQUIED]
        }
    )
);
