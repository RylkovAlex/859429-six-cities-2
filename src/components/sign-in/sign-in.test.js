import React from 'react';
import renderer from 'react-test-renderer';
import {SignIn} from './sign-in';
import {MemoryRouter} from 'react-router';

jest.mock(`../header/header.jsx`, () => () => (<div/>));

it(`SignIn correctly renders after relaunch`, () => {
  const props = {
    setInput: jest.fn(),
    user: null,
    inputValues: {
      email: ``,
      password: ``,
    },
    logIn: jest.fn(),
  };

  const tree = renderer
    .create(
        <MemoryRouter>
          <SignIn {...props}/>
        </MemoryRouter>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
