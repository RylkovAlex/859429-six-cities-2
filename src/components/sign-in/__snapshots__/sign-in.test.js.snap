// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn correctly renders after relaunch 1`] = `
<div
  className="page page--gray page--login"
>
  <div />
  <main
    className="page__main page__main--login"
  >
    <div
      className="page__login-container container"
    >
      <section
        className="login"
      >
        <h1
          className="login__title"
        >
          Sign in
        </h1>
        <form
          className="login__form form"
        >
          <div
            className="login__input-wrapper form__input-wrapper"
          >
            <label
              className="visually-hidden"
            >
              E-mail
            </label>
            <input
              className="login__input form__input"
              name="email"
              onChange={[Function]}
              placeholder="Email"
              required=""
              type="email"
            />
          </div>
          <div
            className="login__input-wrapper form__input-wrapper"
          >
            <label
              className="visually-hidden"
            >
              Password
            </label>
            <input
              className="login__input form__input"
              name="password"
              onChange={[Function]}
              placeholder="Password"
              required=""
              type="password"
            />
          </div>
          <button
            className="login__submit form__submit button"
            disabled={true}
            onClick={[Function]}
            type="submit"
          >
            Sign in
          </button>
        </form>
      </section>
      <section
        className="locations locations--login locations--current"
      >
        <div
          className="locations__item"
        >
          <a
            className="locations__item-link"
            href="/"
            onClick={[Function]}
          >
            <span>
              Amsterdam
            </span>
          </a>
        </div>
      </section>
    </div>
  </main>
</div>
`;
